go/database/mpt/tool/export_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt"
+	cio "github.com/Fantom-foundation/Carmen/go/database/mpt/io"
+)
+
+func TestExport_CanBeInterrupted(t *testing.T) {
+	type testFuncs struct {
+		export   func(context.Context, string, io.Writer) error
+		createDB func(t *testing.T, dir string)
+	}
+	tests := map[string]testFuncs{
+		"archive": {
+			export:   cio.ExportArchive,
+			createDB: createArchiveTrie,
+		},
+		"live": {
+			export:   cio.Export,
+			createDB: createLiveTrie,
+		},
+	}
+
+	for name, tf := range tests {
+		t.Run(name, func(t *testing.T) {
```

Comment:
I think this test is fragile.  What if the inner go function is never scheduled. The test will finish without kill was triggered and it would look like a passing test. 

I think it should work this way:
* `export`  executed in a go routine. 
* only when it is confirmed this routine was scheduled, proceed in the main thread
*  in the main thread wait a bit to make sure export starts to export. 
*  trigger kill signal
*  check the export was not finished

ideally you can utilise your mock writer to count number of writes. This way you can nicely synchronise both threads:
* trigger kill only when `count` >= 1 - it will make sure that export has started
* after kill is processed, check that `count <  maxCount`

Reply:
looking more at the test data, I think there is actually  a huge change that the go routine is never scheduled - the test data is small, i.e. export will run quickly while the go-routine is doing nothing 1s

Reply:
Each write is a one second `Sleep` but your suggestions are very much reasonable. I will rework the test

Reply:
right, I forget that the writing is slow as well.  Still the current code does not assure that the go routine is planned at all, even if it is probably rare for this to happen.